File: Module - Preferences/Preferences/app/src/main/java/com/javadude/preferences/MainActivity.java
package com.javadude.preferences;

import android.content.Intent;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

	private SharedPreferences preferences;
	private TextView nameField;
	private EditText nameEntry;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		nameField = (TextView) findViewById(R.id.nameField);
		nameEntry = (EditText) findViewById(R.id.nameEntry);
		preferences = PreferenceManager.getDefaultSharedPreferences(this);

		String currentName = preferences.getString("some_name2", "No Name");
		nameEntry.setText(currentName);
		nameEntry.addTextChangedListener(new TextWatcher() {
			@Override
			public void beforeTextChanged(CharSequence s, int start, int count, int after) {
			}

			@Override
			public void onTextChanged(CharSequence s, int start, int before, int count) {
			}

			@Override
			public void afterTextChanged(Editable s) {
				SharedPreferences.Editor edit = preferences.edit();
				edit.putString("some_name2", s.toString());
				edit.apply();
			}
		});
	}

	@Override
	protected void onResume() {
		super.onResume();
		String name = preferences.getString("example_text", "Nobody");
		assert nameField != null;
		nameField.setText(name);
	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		getMenuInflater().inflate(R.menu.menu_main, menu);
		return true;
	}

	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		switch(item.getItemId()) {
			case R.id.settings:
				startActivity(new Intent(this, SettingsActivity.class));
				return true;
			default:
				return super.onOptionsItemSelected(item);
		}
	}
}

File: Module - Activities/ACTIVITIES/Activities1/app/src/main/java/com/javadude/activities1/MainActivity.kt
package com.javadude.activities1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.preference.PreferenceManager
import android.view.View
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    var n = 1

//    inner class ButtonListener(val textView : TextView) : View.OnClickListener {
//        override fun onClick(p0: View?) {
//            n++
//            textView.text = n.toString()
//        }
//    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        n = PreferenceManager.getDefaultSharedPreferences(this)
            .getInt("n", 1)

//        if (savedInstanceState != null) {
//            n = savedInstanceState.getInt("n", 1)
//        }
        val textView = findViewById<TextView>(R.id.textView)
        val button = findViewById<Button>(R.id.button)


        textView.text = n.toString()
//        button.setOnClickListener(ButtonListener(textView))
//        button.setOnClickListener(object : View.OnClickListener {
//            override fun onClick(p0: View?) {
//                n++
//                textView.text = n.toString()
//            }
//        })
        button.setOnClickListener {
            n++
            textView.text = n.toString()
        }
    }

//    override fun onSaveInstanceState(outState: Bundle) {
//        super.onSaveInstanceState(outState)
//        outState.putInt("n", n)
//    }

    override fun onPause() {
        super.onPause()
        PreferenceManager.getDefaultSharedPreferences(this)
            .edit()
            .putInt("n", n)
            .apply()
    }
}
